### ISSUE TITLE
Tracking: Use email-first for all flows

### ISSUE THREAD

**Issue by shane-tomlinson** [mozilla/fxa#639]
This will be contentious, and we may need to do some more work to make it so. But, we have globally rolled out email-first to all firstrun users w/o a hit to our retention.

Should we make email-first the one and only way to interface with FxA?

It's a pretty simple plan really, but I see it working like this:

- The email-first screen needs to be a bit more intelligent - [it would need to redirect to the "cached signin page" if the user already has a session.](https://github.com/mozilla/fxa-content-server/issues/6082)
- For OAuth reliers, `action=signup` and `action=signin` would be treated the same as `action=email`, the user would go to `/oauth/`, the "enter your email" screen.
- For Sync, `/signup` and `/signin` would both redirect to `/`, the "enter your email" screen.

Why do I want to do this? So we can stop maintaining so many screens and we can get rid of signin from signup once and for all.

┆Issue is synchronized with this [Jira Epic](https://jira.mozilla.com/browse/FXA-762)
┆Issue Number: FXA-762

**Comment by davismtl** [comment]
I think we should be able to move over to email-first by default but there is a caveat.

I'll try to summarize a Slack conversation we recently had.

@shane-tomlinson recently made our login experience really seamless. As in, if you
1- click on a button to login to a service (like Send)
2- and that you are already signed in to another FxA RP, we would detect that and
3- make login just one more click and you'd be done.

Notice how I started that flow by clicking a login button. If we started with an email first flow where the form was embedded into the page owned by the service, users would have to enter their email first before we detect if they have an account or not.

This means that for email-first to become default and for us to leverage the new seamless login flows, we would need to:
- no longer allow services to embed the email-first form into their pages
- encourage the use of a button

If we encourage the use of a button, we should then explore if the account detection can be done on the button side (to remove 1-click and just be smarter about it). I'm sure there are plenty of security and legal implications there for us to review.

**Comment by rfk** [comment]
> users would have to enter their email first before we detect if they have an account or not.

Would the experience in this case be:

1) User enters their email on the RP side; it happens to match the one that's already signed in to another RP
2) They get directed over to FxA, where we detect that they entered the same email as already signed in and show the "one click" experience

I agree something like [Shane's demo](https://youtu.be/BxTuCh75xAU) would be smoother, but the above doesn't seem *too* bad as an intermediate step IMHO, especially for existing sites that don't want to re-design around changing from email form to a button.

**Comment by davismtl** [comment]
> > users would have to enter their email first before we detect if they have an account or not.
> 
> Would the experience in this case be:
> 
>     1. User enters their email on the RP side; it happens to match the one that's already signed in to another RP
> 
>     2. They get directed over to FxA, where we detect that they entered the same email as already signed in and show the "one click" experience
> 
> 

I think that is one of the flows. Here is what I assume will be the flows in order of harder to easier:
- Email-first flow where form start on RP site
- A button that leads to our email-first flow on accounts.firefox.com
- A smart button like the one in Shane's demo

I think that the first option is completely acceptable but the feedback we got in the #user-journey Slack Channel on September 6th made me realize that in some cases it may make our flows seem a little dump. Here's a thread that popped-up following feedback and the answer Jennifer provided to try to clarify. (you may need to go in channel to get all context)

https://mozilla.slack.com/archives/C4D34NGFL/p1567792344019300

> I agree something like [Shane's demo](https://youtu.be/BxTuCh75xAU) would be smoother, but the above doesn't seem _too_ bad as an intermediate step IMHO, especially for existing sites that don't want to re-design around changing from email form to a button.

I think that's where we'll want to go but just starting with a button could work. We saw how efficient it has been for Monitor since Shane landed the more seamless login.

TLDR; 
1) Yes, let's move to email-first by default. 
2) Let's see how Shane's more seamless login might have improved the email-first flow.

**Comment by davismtl** [comment]
Started doing some analysis of this but the flows are a bit hard to compare...
https://analytics.amplitude.com/mozilla-corp/dashboard/ckwo9zr

**Comment by davismtl** [comment]
When we close this, we should go and close this:
https://bugzilla.mozilla.org/show_bug.cgi?id=1584369

@shane-tomlinson Just to confirm, with the redirect happening, are we making sure to pass entrypoint and utm params to the email-first flow? That was perhaps the only concern the mobile team had. They are optimizing logins and registrations and they want to make sure that metrics don't break.

**Comment by shane-tomlinson** [comment]
> Just to confirm, with the redirect happening, are we making sure to pass entrypoint and utm params to the email-first flow?

It's an internal redirect instead of an HTTP level redirect, the params are propagated:

![utm-params-forwarded](https://user-images.githubusercontent.com/848085/67974040-8d7a0480-fc09-11e9-9211-faf906e3d131.gif)
